src/cpu/rv64/cpu_isa_traits.hpp

```diff
@@ -0,0 +1,118 @@
+/*******************************************************************************
+* Copyright 2018 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#ifndef CPU_RV64_CPU_ISA_TRAITS_HPP
+#define CPU_RV64_CPU_ISA_TRAITS_HPP
+
+#include <fstream>
+#include <string>
+#include <sys/auxv.h>
+#include <type_traits>
+
+#include "common/type_helpers.hpp"
+#include "common/utils.hpp"
+#include "dnnl_types.h"
+
+namespace dnnl {
+namespace impl {
+namespace cpu {
+namespace rv64 {
+
+enum cpu_isa_bit_t : unsigned {
+    v_bit = 1u << 0,
+    zvfh_bit = 1u << 1,
+};
+
+enum cpu_isa_t : unsigned {
+    isa_undef = 0u,
+    v = v_bit,
+    zvfh = zvfh_bit | v,
+    isa_all = ~0u,
+};
+
+struct Riscv64Cpu {
+    bool has_v = false;
+    bool has_zvfh = false;
```

Comment:
I'd suggest to move members into private section and provide interface to query those values, it would provide some more flexibility in the future.